liquidprompt

```diff
@@ -1695,6 +1698,38 @@ else
     _lp_time() { : ; }
 fi
 
+##################################
+# Detect if we're in a container #
+##################################
+
+_lp_container()
+{
+    (( LP_ENABLE_CONTAINER )) ||  return
+
+    _lp_container=""   
+
+    if [[ "$OSTYPE" == "linux-gnu" ]] ; then 
+        if test -e /.dockerenv ||  grep -q "docker" /proc/1/cgroup  ; then
+	    _lp_container="Docker"
+	    
+	elif  grep -q "lxc" /proc/1/cgroup  ; then
+	    _lp_container="LXC"
+	    
+	fi
+	
+	if test -z ${SINGULARITY_CONTAINER+x} || test -z ${SINGULARITY_NAME+x}  ; then
```

Comment:
`test` is less prefered than `[[` forms. All our supported shells support `[[`, use that instead. `[[` does not require quoting variables, while `[` and `test` do.

`-n ${var-}` has less issues than `-z ${var+x}` generally, so use that instead. Unless you are trying to test if the variable is set? I don't see how that would be a case were this should be triggered though.